eZ/Publish/Core/Search/Common/FieldValueMapper/IdentifierMapper.php

```diff
@@ -48,7 +48,7 @@ public function map(Field $field)
      */
     protected function convert($value)
     {
-        // Remove non-printable characters
-        return preg_replace('([^A-Za-z0-9/]+)', '', $value);
+        // Remove characters outside of space and tilde scope (accept range \x20-\x7E)
+        return preg_replace('([^ -~]+)', '', $value);
```

Comment:
this seems a bit weird at 1st sight

Reply:
> this seems a bit weird at 1st sight

Please see https://catonmat.net/my-favorite-regex

Reply:
Haha excellent web page.
You might want to rewrite the comment explaining it though, for dumbasses like me. Suggestion: 'Remove characters outside the space to tilde range, ie. 00 to 31')

Still, my bad feeling about this persists: we are supposed to be stripping away as few characters as possible here, not as many as possible.
For sure we should not be stripping out unicode chars. Does the cute regexp allow them through?
b) it is arguable

Reply:
> Still, my bad feeling about this persists: we are supposed to be stripping away as few characters as possible here, not as many as possible.
> For sure we should not be stripping out unicode chars. Does the cute regexp allow them through?

Are you sure? I don't really see a reason for allowing characters outside of standard ASCII range for **_unique identifiers_**. `ContentService` and `LocationService` allowing everything as `remoteId` is a known technical debt. My plan is to deprecate in eZ Platform 3.1+ such identifiers and do strict validation in eZ Platform 4.0 aligned with the accepted range here.

@gggeek is there a practical use case for remote identifiers containing unicode characters? For special characters (like `:`, `_` and so on) I get it, hence this change. But for e.g. national characters or emojis, I don't think it's valid approach when it comes to identifiers.

Reply:
I think you might be misunderstanding what remote-ids are used for. Besides being used as a handy way to put unique 'labels' on things to fetch which do not vary across environments, they are overwhelmingly used to store an ID coming from a remote system, for 2-way sync. As such there is no guarantee whatsoever about the range of valid characters, and the principle of filtering out as few as possible applies. Why wouldn't a remote system built in china use unicode for its identifiers?

Reply:
Re: "a known technical debt" - I am sorry to come offoften as contrarian, but it seems that our definitions of technical debt are not in mutual agreement...

Reply:
> I think you might be misunderstanding what remote-ids are used for. Besides being used as a handy way to put unique 'labels' on things to fetch which do not vary across environments, they are overwhelmingly used to store an ID coming from a remote system, for 2-way sync. As such there is no guarantee whatsoever about the range of valid characters, and the principle of filtering out as few as possible applies. Why wouldn't a remote system built in china use unicode for its identifiers?

POV ping @andrerom @SylvainGuittard @bdunogier

Reply:
Thank you for your input @gggeek. After internal sync we've decided to handle remote IDs with a dedicated strategy and allow all possible input. We need to purge non-printables and control sequences only, in the similar way like in the case of [StringMapper](https://github.com/ezsystems/ezpublish-kernel/blob/e4e7942bf041ff5128407f81a93af3a95c464e99/eZ/Publish/Core/Search/Common/FieldValueMapper/StringMapper.php#L51-L63). That one is unavoidable as it would break Solr indexing otherwise.

Reply:
:-) cheers